### PR TITLE
ENH: blocking queue size as a custom metric in blocking buffer

### PR DESCRIPTION
**Pull request by chenqi0805**
*Issue #, if available:*

*Description of changes:*
This PR introduces blocking queue size as a gauge in BlockingBuffer.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: data-prepper-api/src/main/java/com/amazon/dataprepper/metrics/MetricNames.java [hunk 1]
```diff
@@ -26,10 +26,20 @@ private MetricNames() {}
...
     /**
-     * Metric representing the number of records read from a buffer
+     * Metric representing the number of records read from a buffer.
      */
     public static final String RECORDS_READ = "recordsRead";
 
+    /**
+     * Metric representing the number of records read from a buffer but unchecked.
+     */
+    public static final String RECORDS_INFLIGHT = "recordsInflight";
```
Comment (laneholloway) [inline data-prepper-api/src/main/java/com/amazon/dataprepper/metrics/MetricNames.java:36]: `recordsInFlight`

**Review by kowshikn — APPROVED** [review]
LGTM. Metrics seems redundant, but I think we can figure out/fix that if required. Is #294 still relavant?

**Comment by chenqi0805** [comment]
@kowshikn This was meant to be reviewed after #294 . I should have add WIP to distinguish. The motivation is to monitor blockingQueue size + recordsInFlight below capacity. I will push some name changes as discussed in slack channel in #294 .

File: data-prepper-api/src/main/java/com/amazon/dataprepper/metrics/MetricNames.java (hunk 1 above)
Comment (chenqi0805) [reply data-prepper-api/src/main/java/com/amazon/dataprepper/metrics/MetricNames.java:36]: Addressed in #294

**Comment by chenqi0805** [comment]
Close the PR as it is redundant with `recordsWritten` - `recordsRead`

**Pull request closed without merging**